controllers/nodefeaturediscovery_status.go

```diff
@@ -282,8 +279,32 @@ func (r *NodeFeatureDiscoveryReconciler) getTopologyUpdaterDaemonSetConditions(c
 
 // getMasterDaemonSetConditions is a wrapper around "getDaemonSetConditions" for
 // master DaemonSets
-func (r *NodeFeatureDiscoveryReconciler) getMasterDaemonSetConditions(ctx context.Context, instance *nfdv1.NodeFeatureDiscovery) (Status, error) {
-	return r.getDaemonSetConditions(ctx, instance, nfdMasterApp)
+func (r *NodeFeatureDiscoveryReconciler) getMasterDeploymentConditions(ctx context.Context, instance *nfdv1.NodeFeatureDiscovery) (Status, error) {
+	return r.getDeploymentConditions(ctx, instance, nfdMasterApp)
+}
+
+// getDaemonSetConditions gets the current status of a DaemonSet. If an error
+// occurs, this function returns the corresponding error message
+func (r *NodeFeatureDiscoveryReconciler) getDeploymentConditions(ctx context.Context, instance *nfdv1.NodeFeatureDiscovery, nfdAppName string) (Status, error) {
+	// Initialize the resource's status to 'Degraded'
+	status := initializeDegradedStatus()
+
+	d, err := r.getDeployment(ctx, instance.ObjectMeta.Namespace, nfdAppName)
+	if err != nil {
+		return status, err
+	}
+
+	dStatus := d.Status.DeepCopy()
+
+	if dStatus.AvailableReplicas == 0 {
```

Comment:
Just a question: is one replica enough? I think in this PR we always use only one replica so it probably is. Maybe add a TODO note to implement more intelligent logic (and more conditions?) if we make the number of replicas configurable.

Reply:
👍🏽  like the idea will add